src/pages/[[...sourceFileId]].tsx

```diff
@@ -0,0 +1,62 @@
+import App from '../App';
+import { GetServerSideProps } from 'next';
+import { gQuery } from '../utils';
+import { GetSourceFileDocument } from '../graphql/GetSourceFile.generated';
+import { SourceRegistryData } from '../types';
+
+export interface SourceFileIdPageProps {
+  sourceRegistryData: SourceRegistryData | null;
+}
+
+export const getServerSideProps: GetServerSideProps<
+  SourceFileIdPageProps,
+  { sourceFileId?: string[] }
+> = async (ctx) => {
+  // dynamic pages always have `req.params` available
+  const sourceFileIdParam = ctx.params!.sourceFileId;
+
+  if (!sourceFileIdParam) {
+    return {
+      props: {
+        sourceRegistryData: null,
+      },
+    };
+  }
```

Comment:
This makes me a little sad that we can't handle this just by a URL - it means that all first loads of the site will have to go through SSR, even though they won't resolve anything. That's some lambda compute time wasted. But I understand why it's here.

Reply:
Yeah, that's right and it saddens me quite a bit as well. OTOH - if we'd be able to use the authenticated user here (which I hope we'll be able to do in the future) then we could SSR user-related parts of the UI, even for paths without an ID.